### PR TITLE
[Builder] Install mlrun with python requirements

### PR DESCRIPTION
**Pull request by alonmr**
We need to install mlrun in the same command as with all other python packages so that pip will take into consideration the mlrun requirements.
Implementation:
1. resolve the required mlrun version
2. add it to the requirements list
3. requirements list is being dumped into the requirement file
4. requirement file is being installed

### CODE REVIEW

File: mlrun/api/utils/builder.py [hunk 1]
```diff
@@ -341,11 +342,13 @@ def build_image(
...
-        mlrun_command = resolve_mlrun_install_command(
+        mlrun_version = resolve_mlrun_install_command_version(
             mlrun_version_specifier, client_version, commands
         )
-        if mlrun_command:
-            commands.append(mlrun_command)
+
+        # mlrun must be installed with other python requirements in the same pip command to avoid version conflicts
+        if mlrun_version:
+            requirements_list = [mlrun_version] + requirements_list
```
Comment (liranbg) [inline mlrun/api/utils/builder.py:351]:
```suggestion
            requirements_list.insert(0, mlrun_version)
```

**Review by liranbg — CHANGES_REQUESTED** [review]
very well! minor comment

File: mlrun/api/utils/builder.py (hunk 1 above)
Comment (alonmr) [reply mlrun/api/utils/builder.py]: funny, that's what I did at first but then I thought that the other way is more readable.
fixed

**Review by liranbg — APPROVED** [review]

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
good pr and nice test coverage overall.
But I think as we already discussed in your previous PR, there is a scenario where a user can pass `with_mlrun=True` but we change it to `with_mlrun=False` if we see that the image is one of ours (`mlrun/mlrun or mlrun/ml-models` etc..). Now I think that we should add the mlrun requirement any way if there are other requirements as well as part of the build so that the requirements will be installed all together.

**Review by theSaarco — COMMENTED** [review]
Looks good, though the `frontend-spec` is changed here, so I'm a bit concerned - why is it in the `frontend-spec` to begin with? Could it be someone is using it and may break due to this change?

**Review by theSaarco — APPROVED** [review]
👍 sababa

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**